File: Data Structures & Algorithms/valid-palindrome-ii/submission-2.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        i = 0
        j = len(s) - 1
        count = 0
        isOdd = len(s) % 2 == 0
        while i < j:
            if s[i] != s[j]:
                count +=1
            i += 1
            j -= 1
        return True if count <= 1 else False

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
      unordered_set<char> set;
      if (s.size() < 1) return 1;
      int left = 0;
      int right = 0;
      int maxS = 1;

      while (right < s.size()){
        while (set.count(s[right])){
          set.erase(s[left]);
          left++;
        }
        set.insert(s[right]);
        right++;
        maxS = max(maxS,right - left);
      }
      return maxS;

    }
};
